packages/generator/src/parser/common/entity-parser.ts

```diff
@@ -0,0 +1,253 @@
+/* Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. */
+import { createLogger, MapType } from '@sap-cloud-sdk/util';
+import { last } from 'rambda';
+import {
+  edmToFieldType,
+  forceArray,
+  isCreatable,
+  isDeletable,
+  isNullableProperty,
+  isUpdatable
+} from '../../generator-utils';
+import {
+  VdmComplexType,
+  VdmEntity,
+  VdmNavigationProperty,
+  VdmProperty
+} from '../../vdm-types';
+import { ServiceNameFormatter } from '../../service-name-formatter';
+import {
+  checkCollectionKind,
+  filterUnknownEdmTypes,
+  isCollection,
+  isComplexType,
+  parseTypeName,
+  propertyJsType
+} from '../util/parser-util';
+import { applyPrefixOnJsConfictParam } from '../../name-formatting-strategies';
+import { SwaggerMetadata } from '../swagger/swagger-types';
+import {
+  entityDescription,
+  propertyDescription
+} from '../util/description-util';
+import {
+  EdmxEntitySetBase,
+  EdmxEntityTypeBase,
+  EdmxNamed,
+  JoinedEntityMetadata
+} from './edmx-types';
+
+const logger = createLogger({
+  package: 'generator',
+  messageContext: 'edmx-entity-parser'
+});
+
+export function parseEntitySetsBase(root): EdmxEntitySetBase[] {
+  return forceArray(root.EntityContainer.EntitySet);
+}
+
+// TODO more elegant way to pass the type in?
+export function parseEntityTypesBase<NavigationType>(
+  root,
+  foo: NavigationType
+): EdmxEntityTypeBase<NavigationType>[] {
```

Comment:
So on the when you parse the base type you do a ` e.NavigationProperty = forceArray(e.NavigationProperty);`. Since `root` has type any this works. However I wanted to indicate in the interface that the property is there and will be parsed. The type however differes between v2 and v4. Hence this generic form.